packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -42,4 +56,28 @@ class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJa
       }
     }
   }
+
+  @ReactMethod
+  fun getSize(url: String, promise: Promise) {
+    try {
+      Glide
+          .with(context)
+          .`as`(ExpoImageSize::class.java)
+          .apply(sizeOptions)
+          .load(url)
+          .into(object : SimpleTarget<ExpoImageSize>() {
+            private val promiseRef = promise
+
+            override fun onResourceReady(resource: ExpoImageSize, transition: Transition<in ExpoImageSize>?) {
+              promiseRef.resolve(resource.asWritableNativeMap())
+            }
+
+            override fun onLoadFailed(errorDrawable: Drawable?) {
+              promiseRef.reject("ERR_IMAGE_GETSIZE_FAILURE", "Failed to get size of the image: $url")
```

Comment:
Is it possible to have a more informative error message? 🤔 
I mean, `getsize failure` is not saying much. Do we know the reason of loading failures? Wrong image metadata? Or the image itself is corrupted? It would be nice to have a least some information, but beware of suggesting something you're not perfectly sure 😉

Reply:
I believe it is not possible in this function to retrieve any information. 
However it is possible to use another mechanism: `RequestListener`--  will try to implement it this way.